src/backend/jitsi_magnify/core/api.py

```diff
@@ -32,42 +35,103 @@ def exception_handler(exc, context):
 
     return drf_exception_handler(exc, context)
 
-class CustomTokenSerializer(TokenObtainPairSerializer):
-    """Api viewset to generate a custom token"""
 
-    @classmethod
-    def get_token(cls, user, room):
-        """Method to get the token created based on a user and a room which they wish to access"""
-        token = super().get_token(user)
+def create_payload(user, room):
+    """Create the payload so that it contains each information jitsi requires"""
+    expiration_seconds = settings.JWT_CONFIGURATION.get("token_expiration_seconds")
 
-        # Add custom claims
-        jitsiUser = {'avatar': settings.JWT_CONFIGURATION.get("guest_avatar"),
-                'name': user.username,
-                'email': user.email}
+    token_payload = {
+        "user_id": user.id,
+        "exp": datetime.datetime.utcnow()
+        + datetime.timedelta(seconds=expiration_seconds),
+        "iat": datetime.datetime.utcnow(),
+        "moderator": user.is_staff,
+        "aud": "jitsi",
+        "iss": settings.JWT_CONFIGURATION.get("jitsi_app_id"),
+        "sub": "meet.jitsi",
+        "room": room,
+    }
 
-        token['context'] = {'user': jitsiUser}
+    jitsi_user = {
+        "avatar": settings.JWT_CONFIGURATION.get("guest_avatar"),
+        "name": user.username,
```

Comment:
@thomas-borot I saw your commit. But it could just be done like this maybe?
```suggestion
        "name": user.username if user.is_authenticated else settings.JWT_CONFIGURATION.get("guest_name"),
```